### ISSUE TITLE
cf target outputs the wrong error message when 502 response is returned

### ISSUE THREAD

**Issue by pdelagrave** [cloudfoundry/cli#1054]
## Command  
```
cf target -s system
FAILED
Unable to access space system.
Space system not found
```
## What occurred
Right after creating a space, we try to target it but the api host is not available, returning 502. The CF CLI client returns 
```
Unable to access space system.
Space system not found
```

Which is very confusing, letting the user think the server was really reached and something is probably wrong with CloudController (as we know for a fact that the space exists and we have the permission to target it).

## What you expected to occur
An error message specifying the cloud controller could not be actually reached instead. 

## CLI Version
Used 6.21.1+6fd3c9f but seems to exists in the master branch as January 11th 2017 (3c307aa3bfc8b7c86d4ba267b1ed9efe01f11b49)

## CC API Endpoint Version  
Doesn't matter, unreachable.

## CF Trace
```
$ CF_TRACE=1 cf target -s system

REQUEST: [2017-01-11T09:35:14-05:00]
GET /v2/organizations/3d2a21b3-7955-49c9-bee7-a6e849262bb3/spaces?q=name%3Asystem&inline-relations-depth=1 HTTP/1.1
Host: api.sys.redacted.com
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Connection: close
Content-Type: application/json
User-Agent: go-cli 6.21.1+6fd3c9f / darwin


FAILED
Unable to access space system.
Space system not found
```

## Platform & Shell Details  
OS X 10.11.6

**Comment by staylor14** [comment]
It's actually even worse than that (now) if you just run `cf-target`(with or without arguments) and you get a 502 with an http response, as you do on, say, GCP:

```
→ cf target -s CATS-persistent-space
Error Code: 502
Raw Response:
<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>502 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>

FAILED
```

This was with cf version 6.23.1+a70deb38f.2017-01-13.

**Comment by dkoper** [comment]
Hi @staylor14 

We are refactoring the `target` command and believe this issue is now addressed, but confirmation is left as an exercise for the issue submitter. 8-p

Please download an [edge binary](https://github.com/cloudfoundry/cli#edge-binaries) and set environment variable CF_CLI_EXPERIMENTAL=true when running the command: `CF_CLI_EXPERIMENTAL=true cf target`.
The first line outputted should be "This command is in EXPERIMENTAL stage and may change without notice".
Please let us know how you go.

Cheers,
Dies Koper
CF CLI PM

**Comment by dkoper** [comment]
I believe we have addressed this in the cf CLI 6.25.0 release.
https://github.com/cloudfoundry/cli#downloads

Cheers,
Dies Koper
CF CLI PM

**Comment by anishp55** [comment]
→ CF_TRACE=1 cf spaces                                                                            JOBS: 0
Getting spaces in org system as admin...


REQUEST: [2017-03-29T14:55:17-04:00]
GET /v2/organizations/3b59992b-dfd8-4bbf-897f-1f984444da9e/spaces?order-by=name&inline-relations-depth=1 HTTP/1.1
Host: api.system.r1.cengage.io
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Connection: close
Content-Type: application/json
User-Agent: go-cli 6.25.0+787326d95.2017-02-28 / darwin



RESPONSE: [2017-03-29T14:55:18-04:00]
HTTP/1.1 500 Internal Server Error
Connection: close
Content-Length: 99
Content-Type: application/json;charset=utf-8
Date: Wed, 29 Mar 2017 18:55:18 GMT
Server: nginx
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: 3897605d-32e3-4757-708f-b7618eb8bc60
X-Vcap-Request-Id: 3897605d-32e3-4757-708f-b7618eb8bc60::4ad0a212-c165-42a6-b1fc-f5bf903a33fd

{
  "error_code": "UnknownError",
  "description": "An unknown error occurred.",
  "code": 10001
}

name
No spaces found

**Comment by dkoper** [comment]
@anishp55 Your case should be resolved in 6.32.0, where we refactored the `spaces` command.
